sdk/tables/Azure.Data.Tables/src/Queryable/ExpressionNormalizer.cs

```diff
@@ -144,6 +188,11 @@ internal Expression VisitMethodCallNoRewrite(MethodCallExpression call)
 
             if (ReflectionUtil.s_dictionaryMethodInfosHash.Contains(visited.Method) && visited.Arguments.Count == 1 && visited.Arguments[0] is ConstantExpression)
             {
+                if (visited.Method.ReturnType == typeof(bool?) && IsImplicitBooleanComparison(visited))
+                {
+                    return CreateExplicitBooleanComparison(visited);
+                }
+
```

Comment:
I don't this this code is ever reached

Reply:
Oh yeah, that's right! I missed revisiting this piece of code after I performed the changes for rewrite expressions that at least has a type casting expression on it. That said, the type casting rewrites will cover the scenario when having expressions such as `ent => (bool)ent.GetBoolean("BoolProp")`.